code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -347,13 +358,24 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 	if(Adjacent(user))
 		. += span_notice("Paper level: [stored_paper] / [max_paper].")
 
-/obj/item/modular_computer/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+/obj/item/modular_computer/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
 	. = ..()
 
-	context[SCREENTIP_CONTEXT_ALT_LMB] = "Remove ID"
-	context[SCREENTIP_CONTEXT_CTRL_SHIFT_LMB] = "Remove Disk"
+	var/obj/item/computer_hardware/card_slot/card_slot = all_components[MC_CARD]
+	var/obj/item/computer_hardware/card_slot/card_slot2 = all_components[MC_CARD2]
+
+	if(card_slot?.stored_card || card_slot2?.stored_card) // IDs get removed first before pAIs
+		context[SCREENTIP_CONTEXT_ALT_LMB] = "Remove ID"
+	else if(inserted_pai)
+		context[SCREENTIP_CONTEXT_ALT_LMB] = "Remove pAI"
 
-	return CONTEXTUAL_SCREENTIP_SET
+	if(all_components[MC_SDD])
+		context[SCREENTIP_CONTEXT_CTRL_SHIFT_LMB] = "Remove SSD"
+
+	if(all_components[MC_SDD] || card_slot?.stored_card || card_slot2?.stored_card || inserted_pai)
```

Comment:
Instead of doing this, which can degrade over time as more changes are made, can you have every path that sets the context also do `. = CONTEXTUAL_SCREENTIP_SET`, then `return .` at the end?

Reply:
Applied with https://github.com/tgstation/tgstation/pull/68251/commits/2623e8967678f1701ec41c3e80bed8ba9f8c83a3